ui/src/app/workflows/components/workflows-list/workflows-list.tsx

```diff
@@ -212,6 +244,12 @@ export function WorkflowsList({match, location, history}: RouteComponentProps<an
                                 setFinishedBefore(date);
                                 clearSelectedWorkflows(); // date filters are client-side, but clear similar to the server-side ones for consistency
                             }}
+                            name={name}
+                            setName={setName}
+                            namePrefix={namePrefix}
+                            setNamePrefix={setNamePrefix}
+                            namePattern={namePattern}
+                            setNamePattern={setNamePattern}
```

Comment:
I don't think this is actually necessary -- as you wrote, only 1 will be set at a time and they are all strings. We can just keep state for a single variable and query differently depending on what is chosen. E.g. `nameValue` or `nameInput`